xdsl/interactive/app.py

```diff
@@ -174,6 +174,8 @@ def action_quit_app(self) -> None:
     def on_clear_input_button_pressed(self, event: Button.Pressed) -> None:
         """When the "Clear Input" button is pressed, the input IR TextArea is cleared and the current_module is updated"""
         self.input_text_area.clear()
+        # when input is cleared, deselect the passes
+        self.passes_selection_list.deselect_all()
```

Comment:
that's pretty unexpected behaviour, why should the selected passes change?

Reply:
If you were using this app, not having built it, would you expect a button that says "Clear Input" to do something other than clearing the input? But in general it is common to want to apply a certain pass pipeline to different inputs